### PR TITLE
Implement testCreateTable and testDropTable in Kudu connector test

### PR DESCRIPTION
**Pull request by chen-ni**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

An improvement in testing.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Connector tests.

> How would you describe this change to a non-technical end-user or system administrator?

Improved some tests that shouldn't change any behavior.

## Related issues, pull requests, and links
A series of PRs will be created to implement https://github.com/trinodb/trino/issues/11815. This is the first one.

There are 14 tests to fix in total, this PR fixes the first 2.

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

**Comment by chen-ni** [comment]
Just emailed the signed CLA. Will the check automatically pass when the document is received?

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java [hunk 1]
```diff
@@ -272,8 +275,55 @@ public void testExplainAnalyzeWithDeleteWithSubquery()
...
+        String tableName = "test_create_" + randomTableSuffix();
+        if (!hasBehavior(SUPPORTS_CREATE_TABLE)) {
+            assertQueryFails(
+                    "CREATE TABLE " + tableName + " (" +
+                            "id INT WITH (primary_key=true)," +
+                            "a bigint, b double, c varchar(50))" +
+                    "WITH (partition_by_hash_columns = ARRAY['id'], partition_by_hash_buckets = 2)",
+                    "This connector does not support creating tables");
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (chen-ni) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java]: @ebyhr Thanks! Fixed in https://github.com/trinodb/trino/pull/12899/commits/07a70bfbf5a4a2ca985901005932757e27710e27 :)

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java [hunk 2]
```diff
@@ -272,8 +275,55 @@ public void testExplainAnalyzeWithDeleteWithSubquery()
...
+
+        assertQueryFails("CREATE TABLE " + tableName + " (" +
+                        "id INT WITH (primary_key=true)," +
+                        "a bad_type, b double, c varchar(50))" +
+                        "WITH (partition_by_hash_columns = ARRAY['id'], partition_by_hash_buckets = 2)",
+                ".* Unknown type 'bad_type' for column 'a'");
+        assertFalse(getQueryRunner().tableExists(getSession(), tableName));
+
+        // TODO (https://github.com/trinodb/trino/issues/5901) revert to longer name when Oracle version is updated
+        tableName = "test_cr_not_exists_" + randomTableSuffix();
```
[the remark this answers is not in the document]
Comment (chen-ni) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java]: Thanks! Fixed in https://github.com/trinodb/trino/pull/12899/commits/05f45f330fe86b3d94657010f224e2f4903a5b90 :)

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java [hunk 3]
```diff
@@ -295,9 +345,17 @@ public void testCreateTableWithColumnComment()
...
-        assertThatThrownBy(super::testDropTable)
-                .hasMessage("Table partitioning must be specified using setRangePartitionColumns or addHashPartitions");
-        throw new SkipException("TODO Enable the test once Kudu connector can create tables with default partitions");
+        // TODO Remove this overriding test once kudu connector can create tables with default partitions
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE));
+        String tableName = "test_drop_table_" + randomTableSuffix();
+        assertUpdate(
+                "CREATE TABLE " + tableName + "(" +
+                    "id INT WITH (primary_key=true)," +
+                    "col bigint)");
```
[the remark this answers is not in the document]
Comment (chen-ni) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java]: @ebyhr Thanks! Yeah I forgot this one.. Fixed in https://github.com/trinodb/trino/pull/12899/commits/3ab65b5075b1c1b8012efb3e8ec2aa98c69b3306 :)

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java [hunk 4]
```diff
@@ -295,9 +345,17 @@ public void testCreateTableWithColumnComment()
     @Override
     public void testDropTable()
     {
-        assertThatThrownBy(super::testDropTable)
-                .hasMessage("Table partitioning must be specified using setRangePartitionColumns or addHashPartitions");
-        throw new SkipException("TODO Enable the test once Kudu connector can create tables with default partitions");
+        // TODO Remove this overriding test once kudu connector can create tables with default partitions
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE));
```
[the remark this answers is not in the document]
Comment (chen-ni) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java]: Thanks! Fixed in https://github.com/trinodb/trino/pull/12899/commits/111c8d3e73ed235f1631bd622bfe4dfd40d5859c :)

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java [hunk 5]
```diff
@@ -272,8 +275,55 @@ public void testExplainAnalyzeWithDeleteWithSubquery()
...
+        assertTrue(getQueryRunner().tableExists(getSession(), tableName));
+        assertTableColumnNames(tableName, "id", "a", "b", "c");
+
+        assertUpdate(
+                "CREATE TABLE IF NOT EXISTS " + tableName + " (" +
+                    "id INT WITH (primary_key=true)," +
+                    "d bigint, e varchar(50))" +
+                    "WITH (partition_by_hash_columns = ARRAY['id'], partition_by_hash_buckets = 2)");
+        assertTrue(getQueryRunner().tableExists(getSession(), tableName));
+        assertTableColumnNames(tableName, "id", "a", "b", "c");
```
[the remark this answers is not in the document]
Comment (chen-ni) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java:313]: @ebyhr I tried, but Kudu doesn't seem to support `CREATE TABLE LIKE`.

Here is the error I got:
```
java.lang.RuntimeException: This connector does not support creating tables with column comment
```

while executing this line:
```java
assertUpdate("CREATE TABLE " + tableNameLike + " (LIKE " + tableName + ", " +
            "d bigint, e varchar(50))" +
            "WITH (partition_by_hash_columns = ARRAY['id'], partition_by_hash_buckets = 2)");
```

However, Kudu does support the `CREATE TABLE …​ AS SELECT ...` syntax: https://kudu.apache.org/docs/kudu_impala_integration.html#_create_table_as_select, should I use this?

[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java:313]: The connector supports the syntax, but it has an issue which will be fixed in #12469. 
Please add a failure case using `assertThatThrownBy()`.

[the remark this answers is not in the document]
Comment (chen-ni) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java:312]: @ebyhr Fixed in https://github.com/trinodb/trino/pull/12899/commits/75c2ff020d8269a2c49c3a6b80ed9327f54f0c76 :)

**Review by ebyhr — APPROVED** [review]
Please squash commits into one. You can update the commit title as "Implement testCreateTable and testDropTable in Kudu connector test". 
https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#git-merge-strategy

**Comment by chen-ni** [comment]
@ebyhr Got it :) Just squashed and force-pushed the commit. Still have to wait for the CLA, though..

**Comment by ebyhr** [comment]
Could you fix checkstyle failure? https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#code-style
```
Error:  src/test/java/io/trino/plugin/kudu/AbstractKuduConnectorTest.java:[39,1] (imports) ImportOrder: Wrong order for 'org.testng.Assert.assertFalse' import.
```

**Comment by chen-ni** [comment]
@ebyhr Sorry, I've been having trouble running the style check locally.. 
Will fix this today after work :)

**Comment by chen-ni** [comment]
Just fixed the checkstyle error, stashed and pushed the commit :)

**Comment by mosabua** [comment]
@ebyhr CLA is done now .. I am not sure if the test failure is a false alarm .. please proceed to review or merge or whatever appropriate as next step. Thanks

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**